File: backend/app/schemas/daily_workflow.py
import enum
import uuid

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.daily_task_generation import DailyTaskGenerationResponse


class DailyWorkflowStatus(str, enum.Enum):
    ACTION_REQUIRED = "ACTION_REQUIRED"
    READY = "READY"


class DailyWorkflowResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    workspace_id: uuid.UUID
    user_id: uuid.UUID
    workflow_date: date
    workflow_status: DailyWorkflowStatus
    form_required: bool
    form_submitted: bool
    definition_id: uuid.UUID | None
    submission_id: uuid.UUID | None
    task_generation: DailyTaskGenerationResponse
    evaluated_at: datetime

File: frontend/src/types/dailyWorkflow.ts
export type DailyWorkflowStatus = "READY" | "ACTION_REQUIRED";

export interface DailyTaskGeneration {
  workspace_id: string;
  generation_date: string;
  eligible_series_count: number;
  created_task_count: number;
  skipped_existing_count: number;
  created_task_ids: string[];
  generated_at: string;
}

export interface DailyWorkflow {
  workspace_id: string;
  user_id: string;
  workflow_date: string;
  workflow_status: DailyWorkflowStatus;
  form_required: boolean;
  form_submitted: boolean;
  definition_id: string | null;
  submission_id: string | null;
  task_generation: DailyTaskGeneration;
  evaluated_at: string;
}

export interface WorkspaceSettings {
  id: string;
  workspace_id: string;
  timezone: string;
  daily_form_enabled: boolean;
  daily_form_reminder_time: string;
  daily_task_generation_enabled: boolean;
  week_starts_on: string;
  created_at: string;
  updated_at: string;
}
